File: 2.0/java/texts/findTextsSmsChat.java
import com.callfire.api.client.CallfireClient;
import com.callfire.api.client.api.callstexts.model.Text;
import com.callfire.api.client.api.callstexts.model.request.FindTextsRequest;
import com.callfire.api.client.api.common.model.Page;

import java.util.Calendar;
import java.util.GregorianCalendar;

class ApiClientSample {
    public static void main(String[] args) {
        CallfireClient client = new CallfireClient("api_login", "api_password");
        FindTextsRequest request = FindTextsRequest.create()
            .fromNumber("12135551100")
            .intervalBegin(new GregorianCalendar(2016, Calendar.SEPTEMBER, 9, 15, 50, 17).getTime())
            .intervalEnd(new GregorianCalendar(2016, Calendar.DECEMBER, 10, 0, 0, 0).getTime())
            .offset(100L)
            .limit(50L)
            .fields("items(id,message,created)")
            .build();
        Page<Text> texts = client.textsApi().find(request);
    }
}

File: 2.0/csharp/texts/findTextsSmsChat.cs
using System;
using CallfireApiClient;
using CallfireApiClient.Api.CallsTexts.Model;
using CallfireApiClient.Api.CallsTexts.Model.Request;
using CallfireApiClient.Api.Common.Model;

public class ApiClientSample
{
    public static void Main(string[] args)
    {
        var client = new CallfireClient("api_login", "api_password");
        var request = new FindTextsRequest
        {
            FromNumber = "12135551100",
            IntervalBegin = new DateTime(2016, 9, 13, 15, 50, 17),
            IntervalEnd = new DateTime(2016, 12, 1, 0, 0, 0),
            Offset = 100,
            Limit = 50,
            Fields = "items(id,message,created)"
        };
        Page<Text> texts = client.TextsApi.Find(request);
    }
}
